packages/core/package.json

```diff
@@ -6,7 +6,12 @@
   "types": "dist/index.d.ts",
   "scripts": {
     "pretest": "npm run lint",
+    "redis": "docker rm --force redis-commerce-sdk; docker run -dp 6379:6379 --name redis-commerce-sdk redis:5-alpine",
+    "redis:shutdown": "docker rm --force redis-commerce-sdk; docker rm --force redis-commerce-sdk-debug",
+    "redis:debug": "docker rm --force redis-commerce-sdk-debug; docker run -dp 6379:6379 --name redis-commerce-sdk-debug redis:5-alpine && docker exec -it redis-commerce-sdk-debug redis-cli monitor",
     "test": "nyc mocha \"test/**/*.test.ts\" --reporter=xunit --reporter-options output=../../reports_integration/core.xml",
+    "test:cache": "nyc --check-coverage=false mocha \"test_integration/**/*.ts\" --reporter=xunit --reporter-options output=../../reports/core-integration.xml --retries 2 --timeout 5000",
+    "test:integration": "npm run test:cache || (echo 'test:cache failed' && echo 'Cache container tail 100 debug logs\n----------------------------\nA complete log of this test:cache run can be found in:testCacheDebug.log\n----------------------------\n' && (docker logs redis-commerce-sdk > ../../testCacheDebug.log || docker logs redis-commerce-sdk-debug --tail 50 > ../../testCacheDebug.log) && (docker logs redis-commerce-sdk --tail 50 || docker logs redis-commerce-sdk-debug --tail 50) && exit 1)",
```

Comment:
This is pretty long, I'd pull it into a shell script and call that. Also, this dumps the docker logs of the container which won't give much useful info since redis runs quietly to stdout/err by default. We want to find the location of the redis logs in the container and dump those with a docker exec tail or a docker cp. Let me know if you want to talk about this.

Reply:
True, I was very close in creating a separate script and parse the failure from logs. I think this is a good start, sure, we can discuss about better options.